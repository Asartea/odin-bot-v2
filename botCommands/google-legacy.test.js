const command = require('./google-legacy');

describe('!google', () => {
  describe('regex', () => {
    it.each([
      ['!google query'],
      [' !google query'],
      ['!google query @odin-bot'],
      ['@odin-bot !google query'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['!googl'],
      ['google'],
      ['!goog'],
      ['!googless'],
      ['```function("!google", () => {}```'],
      ['!googles'],
      [''],
      [' '],
      [' !'],
      ['@odin-bot ! google'],
      ['!goo&gle'],
      ['!goo^gle'],
      ['!google!'],
      ['@odin-bot! google'],
      ['https:!!google.com'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! !google query'],
      ["Don't worry about !google query"],
      ['Hey @odin-bot, !google query'],
      ['!@odin-bot ^ !me !google !tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user!google'],
      ["it's about!google"],
      ['!googleisanillusion'],
      ['!google!'],
      ['!google*'],
      ['!google...'],
    ])(
      "'%s' - command should be its own word!group - no leading or trailing characters",
      (string) => {
        expect(command.regex.test(string)).toBeFalsy();
      },
    );
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(
        command.cb({ content: '!google The Odin Project' }),
      ).toMatchSnapshot();
    });
  });
});
