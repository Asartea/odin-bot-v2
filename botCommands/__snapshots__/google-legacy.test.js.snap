// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!google callback returns correct output 1`] = `
{
  "embeds": [
    {
      "color": 13407555,
      "description": "This [Google query for "The Odin Project"](https://www.google.com/search?q=The+Odin+Project) might help you find what you're looking for.",
      "title": "Search Google",
    },
  ],
}
`;
